src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
+    var syndromePolyPlusOne = syndromePoly.slice(0);
+    syndromePolyPlusOne[syndromePoly.length - 1] ^= 1;
+
+    // Initial conditions
+    var sigmas = new Array(n - k + 1);
+    var omegas = new Array(n - k + 1);
+    var taus = new Array(n - k + 1);
+    var gammas = new Array(n - k + 1);
+    var Ds = new Array(n - k + 1);
+    var Bs = new Array(n - k + 1);
+    sigmas[0] = new Uint8Array([1]);
+    omegas[0] = new Uint8Array([1]);
+    taus[0] = new Uint8Array([1]);
+    gammas[0] = new Uint8Array([0]);
+    Ds[0] = 0;
+    Bs[0] = false;
+
+    // Perform n-k iterations
+    var lhs;
+    var delta;
+    var shiftedTau;
+    var shiftedGamma;
+    var scaledTau;
+    var scaledGamma;
+    var DThreshold;
+    for (var i = 0; i <= n - k; i++) {
+      // Compute the left-hand side of the key equation
+      lhs = UTIL.polynomialMult(sigmas[i], syndromePolyPlusOne);
+
+      // Delta is the coefficient of z^(i+1)
+      delta = lhs[(lhs.length - 1) - (i + 1)];
+
+      // Use tau and gamma to update sigma and omega:
+      shiftedTau = UTIL.mergeTypedArrays(taus[i], new Uint8Array([0])); // z * tau
+      shiftedGamma = UTIL.mergeTypedArrays(gammas[i], new Uint8Array([0])); // z * gamma
+      scaledTau = UTIL.polynomialScale(shiftedTau, delta); // delta * z * tau
+      scaledGamma = UTIL.polynomialScale(shiftedGamma, delta); // delta * z * gamma
+      sigmas[i + 1] = UTIL.polynomialAdd(sigmas[i], scaledTau);
+      omegas[i + 1] = UTIL.polynomialAdd(omegas[i], scaledGamma);
+
+      // Follow the B-M iteration rules to update tau and gamma:
+      DThreshold = (i + 1) / 2;
+      if (delta == 0 || Ds[i] > DThreshold || (Ds[i] == DThreshold && !Bs[i])) {
+        // RULE A
+        taus[i + 1] = UTIL.mergeTypedArrays(taus[i], new Uint8Array([0]));
+        gammas[i + 1] = UTIL.mergeTypedArrays(gammas[i], new Uint8Array([0]));
+      } else {
+        // RULE B
+        var invDelta = UTIL.multInv(delta);
+        taus[i + 1] = UTIL.polynomialScale(sigmas[i], invDelta);
+        gammas[i + 1] = UTIL.polynomialScale(omegas[i], invDelta);
+
+        // Update the D tracker
+        Ds[i + 1] = i + 1 - Ds[i];
+
+        // If the equality case triggered, turn the B flag off
+        if (Ds[i] == DThreshold) {
+          Bs[i + 1] = false;
+        }
+      }
+    }
+
+    return {
+      errorLocationPoly: sigmas[n - k],
+      errorMagnitudePoly: omegas[n - k]
+    };
+  }
+
+  // Performs a Chien search on an error location polynomial to find the set of
+  // error locations.
+  // Input errorLocationPoly is a big-endian array representing a polynomial,
+  // i.e. [1,2,3] -> x^2 + 2x + 3
+  // Output is a set of integers in GF(256), each of which is an error location, GEN^power
+  // -> i.e. taking log3 of an element in the set would give you the error position,
+  // the power at which the error occurred.
+  function chienSearch(errorLocationPoly) {
```

Comment:
you're right ;) I was gonna refactor so this would actually be a Chien search -- will change method name for now though